feat: Adds support for streams accountDetails endpoint (#3364)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/stream_account_details.md

> # Data Source: mongodbatlas_stream_account_details

Reviewer: I am not sure calling it _details is right. I think it should be just called `mongodbatlas_stream_account` - @AgustinBettati @lantoli @EspenAlbert @oarbusi  thoughts?

Reviewer 2: No strong opinion, the path `https://cloud.mongodb.com/api/atlas/v2/groups/{groupId}/streams/accountDetails` and operation id `getAccountDetails` use this term but I can see how it is redundant

Reviewer: fair point, we should stick to the API name (even though I don't really like it)

Reviewer 3: i agree to stick to API although i don't like either, also it can be confused with plural data sources

Author: agreed on the confusing naming, will stick with the API but noting this as we audit the state of our existing apis

> `mongodbatlas_stream_account_details` returns the Account ID, and the VPC ID for the group and region specified.

Reviewer 4: we should clarify that these are AWS IDs. also tweak slightly to remove passive voice.
```suggestion
`mongodbatlas_stream_account_details` returns the AWS Account ID and the AWS VPC ID for the group and region that you specify.
```

## Example Usages
```terraform
data "mongodbatlas_stream_account_details" "account_details" {
  project_id     = var.project_id
  cloud_provider = "aws"
  region_name    = "US_EAST_1"
}

output "aws_account_id" {
  value = data.mongodbatlas_stream_account_details.account_details.aws_account_id
}

output "cidr_block" {
  value = data.mongodbatlas_stream_account_details.account_details.cidr_block
}

output "cloud_provider" {
  value = data.mongodbatlas_stream_account_details.account_details.cloud_provider
}

output "vpc_id" {
  value = data.mongodbatlas_stream_account_details.account_details.vpc_id
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

> - `cloud_provider` (String) One of "aws", "azure" or "gcp".

Reviewer 4: use monospace for field values instead of quotes. 
```suggestion
- `cloud_provider` (String) One of `aws`, `azure`, or `gcp`.
```

> - `project_id` (String) Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.

Reviewer 4: I don't think this relative link will work from the terraform provider docs. use a link like the one on L49

Author: Yep, miss on my end for not replacing these, fixed!

**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.

> - `region_name` (String) The cloud provider specific region name, i.e. "US_EAST_1" for cloud provider "aws".

Reviewer 4: ```suggestion
- `region_name` (String) The cloud provider specific region name, i.e. `US_EAST_1` for cloud provider `aws`.
```

### Read-Only

> 

Reviewer 4: question: some of these are intuitive enough (like azure_subscription_id), but do these other fields apply only for specific cloud providers? If so, it might help to provide that context in the descriptions.

Author: Good question, I made some changes to provide a bit more context. The fields returned depend on the cloud provider so `virtual_network_name` and `virtual_network_name` are only present in the response for Azure while `aws_account_id` and `vpc_id` will be present for AWS

- `aws_account_id` (String) The AWS Account ID.
- `azure_subscription_id` (String) The Azure Subscription ID.
- `cidr_block` (String) The VPC CIDR Block.
- `virtual_network_name` (String) The name of the virtual network.
- `vpc_id` (String) The VPC ID.

> For more information see: [MongoDB Atlas API - Returns the Account ID, and the VPC ID for the group and region specified](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams/operation/getAccountDetails) Documentation.

Reviewer 4: ```suggestion
For more information see [MongoDB Atlas API - Returns the Account ID, and the VPC ID for the group and region specified](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams/operation/getAccountDetails).
```

Reviewer 4: The API name is a bit verbose. My recommendation would be to change that upstream to something that aligns with the style of the other endpoint names.

Author: I agree and we have added this to our list of potential changes in the next version of our API suite, though since this endpoint is used in prod, I don't believe we can update that at this time
